Source/swiftlint-dev/Rules+Register.swift

```diff
@@ -74,53 +75,174 @@ struct NewRuleDetails: Equatable {
+                deps = [
+                    ":TestHelpers",
+                ],
+                copts = copts + strict_concurrency_copts,
+            )
+
+            swift_test(
+                name = "GeneratedTests_{}".format(shard_number),
+                visibility = ["//visibility:public"],
+                deps = [":GeneratedTests_{}.library".format(shard_number)],
+            )
+
+        def generated_tests(copts, strict_concurrency_copts):
+            """Creates all generated test targets for SwiftLint rules.
+
+            Args:
+                copts: Common compiler options
+                strict_concurrency_copts: Strict concurrency compiler options
+            """
+        \#(macroInvocations)
+
+        """#
+    }
+
     func registerInRulesList(_ ruleFiles: [String]) throws {
-        let rules = ruleFiles
+        let rulesImportString = ruleFiles
             .map { $0.replacingOccurrences(of: ".swift", with: ".self") }
             .joined(separator: ",\n")
         let builtInRulesFile = rulesDirectory.deletingLastPathComponent()
             .appendingPathComponent("Models", isDirectory: true)
             .appendingPathComponent("BuiltInRules.swift", isDirectory: false)
-        try """
-            // GENERATED FILE. DO NOT EDIT!
-
-            /// The rule list containing all available rules built into SwiftLint.
-            public let builtInRules: [any Rule.Type] = [
-            \(rules.indent(by: 4)),
-            ]
 
-            """.write(to: builtInRulesFile, atomically: true, encoding: .utf8)
+        let fileContent = generateBuiltInRulesFileContent(rulesImportList: rulesImportString)
+        try fileContent.write(to: builtInRulesFile, atomically: true, encoding: .utf8)
     }
 
     func registerInTests(_ ruleFiles: [String]) throws {
-        let testFile = testsDirectory
-            .appendingPathComponent("GeneratedTests.swift", isDirectory: false)
-        let rules = ruleFiles
-            .map { $0.replacingOccurrences(of: ".swift", with: "") }
-            .map { testName in """
+        let rulesContext = ProcessedRulesContext(ruleFiles: ruleFiles, shardSize: Self.shardSize)
+        let ruleNames = rulesContext.baseRuleNames
+        let totalShards = rulesContext.totalShards
+
+        // Remove old generated files
+        let existingFiles = try FileManager.default.contentsOfDirectory(
+            at: testsDirectory,
+            includingPropertiesForKeys: nil
+        )
+        for file in existingFiles where file.lastPathComponent.hasPrefix("GeneratedTests") &&
+            file.pathExtension == "swift" {
+            try FileManager.default.removeItem(at: file)
+        }
+
+        // Create sharded test files
+        for shardIndex in 0..<totalShards {
+            let startIndex = shardIndex * Self.shardSize
+            let endIndex = min(startIndex + Self.shardSize, ruleNames.count)
+            let shardRules = Array(ruleNames[startIndex..<endIndex])
```

Comment:
Could be a method of `ProcessedRulesContext`, like `shardRule(forIndex shardIndex: Int)`

Reply:
e3b53fe1005351d9ced4a760116a84cb1e4e923b